### PR TITLE
docs update for #1578

### LINKED ISSUE #1584: attachToDocument deprecated and no documented substitute?
### Subject of the issue

attachToDocument was deprecated, in favor of attachTo, which is supposed to be more flexible, by allowing any css selector as the mount point.

But what is the actual replacement for attachToDocument: true? attachTo: document.documentElement yields HierarchyRequestError: The operation would yield an incorrect node tree.

Seems like the functionality of attachToDocument was deprecated in favor of something that does not substitute it, but only provides complementary behaviour.

Note that, besides the hierarchy problem, attachTo should be acompanied by wrapper.destroy() at the end. The thing is that this DESTROYS THE NODE to which the wrapper was attached. So, I can attach the element to the body, but the body is going to be destroyed in the call.

What I'm currently doing is adding a dumb routine to each test suite that needs the wrapper to be attached to the dom:
```

    var node = document.createElement("div");
    node.setAttribute('id', 'app')
    document.body.appendChild(node);

    const wrapper = mount(List, {
      attachTo: '#app'
    }

    ...

    wrapper.destroy() 
```

Needless to say, unless I'm missing something, it would be way easier to not have attachToDocument deprecated.

### PR DESCRIPTION
**Pull request by alexmiddeleer**
Closes #1584

**The PR fulfills these requirements:**

- [ X ] It's submitted to the `dev` branch.
- [ X ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [  ] All tests are passing: https://github.com/vuejs/vue-test-utils/blob/dev/.github/CONTRIBUTING.md#development-setup
- [  ] New/updated tests are included

Note that prettier found some issues in:
docs/zh/guides/common-tips.md
docs/zh/guides/getting-started.md

But I did not touch these files.

image

### CODE REVIEW

**Comment by alexmiddeleer** [comment]
I attempted to fix the pre-existing prettier errors in https://github.com/vuejs/vue-test-utils/pull/1581

File: docs/api/options.md [hunk 1]
```diff
@@ -323,7 +323,18 @@ wrapper.destroy()
 - default: `false`
 
 ::: warning Deprecation warning
-`attachToDocument` is deprecated and will be removed in future releases. Use [`attachTo`](#attachto) instead.
+`attachToDocument` is deprecated and will be removed in future releases. Use [`attachTo`](#attachto) instead. Example:
```
Comment (afontcu) [inline docs/api/options.md]: Could we add a very simple explanation of what does the example do? Say, "For example, if you need to attach the component to the `document.body`" or similar

**Comment by alexmiddeleer** [comment]
Please let me know if there's anything you're still waiting on me for.

**Comment by afontcu** [comment]
Looks good! There are some issues with Prettier. Would you mind running `yarn format` and push again? (I think they might be unrelated to your changes)

**Comment by alexmiddeleer** [comment]
@afontcu I ran `yarn format`, but it resulted in the same code changes that I addressed in https://github.com/vuejs/vue-test-utils/pull/1581. Those changes broke a desired convention in the /zh language docs.

**Comment by afontcu** [comment]
Hm, weird. It def looks like it is related to `zh` docs. Will take a look, but otherwise your PR looks fine 👍  gonna try to find out what's going on and then merge it up.

thanks!! 🤗

**Comment by afontcu** [comment]
Closing in favor of #1611! thanks for this :)

**Pull request closed without merging**